instrumentation/opentelemetry_beam/src/opentelemetry_vm_system_info.erl

```diff
@@ -0,0 +1,291 @@
+%%   </li>
+%%   <li>
+%%     `erlang.vm.ets_limit'<br/>
+%%     Type: gauge.<br/>
+%%     The maximum number of ETS tables allowed.
+%%   </li>
+%%   <li>
+%%     `erlang.vm.logical_processors'<br/>
+%%     Type: gauge.<br/>
+%%     The detected number of logical processors configured in the system.
+%%   </li>
+%%   <li>
+%%     `erlang.vm.logical_processors_available'<br/>
+%%     Type: gauge.<br/>
+%%     The detected number of logical processors
+%%     available to the Erlang runtime system.
+%%   </li>
+%%   <li>
+%%     `erlang.vm.logical_processors_online'<br/>
+%%     Type: gauge.<br/>
+%%     The detected number of logical processors online on the system.
+%%   </li>
+%%   <li>
+%%     `erlang.vm.port_count'<br/>
+%%     Type: gauge.<br/>
+%%     The number of ports currently existing at the local node.
+%%   </li>
+%%   <li>
+%%     `erlang.vm.port_limit'<br/>
+%%     Type: gauge.<br/>
+%%     The maximum number of simultaneously existing ports at the local node.
+%%   </li>
+%%   <li>
+%%     `erlang.vm.process_count'<br/>
+%%     Type: gauge.<br/>
+%%     The number of processes currently existing at the local node.
+%%   </li>
+%%   <li>
+%%     `erlang.vm.process_limit'<br/>
+%%     Type: gauge.<br/>
+%%     The maximum number of simultaneously existing processes
+%%     at the local node.
+%%   </li>
+%%   <li>
+%%     `erlang.vm.schedulers'<br/>
+%%     Type: gauge.<br/>
+%%     The number of scheduler threads used by the emulator.
+%%   </li>
+%%   <li>
+%%     `erlang.vm.schedulers_online'<br/>
+%%     Type: gauge.<br/>
+%%     The number of schedulers online.
+%%   </li>
+%%   <li>
+%%     `erlang.vm.thread_pool_size'<br/>
+%%     Type: gauge.<br/>
+%%     The number of async threads in the async thread pool
+%%     used for asynchronous driver calls.
+%%   </li>
+%%   <li>
+%%     `erlang.vm.wordsize'<br/>
+%%     Type: gauge.<br/>
+%%     The size of Erlang term words in bytes.
+%%   </li>
+%%   <li>
+%%     `erlang.vm.atom_count'<br/>
+%%     Type: gauge.<br/>
+%%     The number of atom currently existing at the local node.
+%%   </li>
+%%   <li>
+%%     `erlang.vm.atom_limit'<br/>
+%%     Type: gauge.<br/>
+%%     The maximum number of simultaneously existing atom at the local node.
+%%   </li>
+%%   <li>
+%%     `erlang.vm.allocators'<br/>
+%%     Type: gauge.<br/>
+%%     Allocated (carriers_size) and used (blocks_size) memory
+%%     for the different allocators in the VM. See erts_alloc(3).
+%%   </li>
+%% </ul>
+%%
+%% @end
+-module(opentelemetry_vm_system_info).
+
+-export([setup/0, setup/1]).
+
+-include_lib("opentelemetry_api_experimental/include/otel_meter.hrl").
+
+%%%===================================================================
+%%% API
+%%%===================================================================
+
+setup() ->
+    setup(?current_meter).
+
+setup(Meter) ->
+    Instruments =
+        [otel_meter:create_observable_gauge(
```

Comment:
You can have more than 1 instrument per callback.

Maybe they shouldn't all be in 1 since then we can't skip the whole callback if a user doesn't want a specific metric but should at least group together ones that will either likely get used together (user can still filter, they'll just all be read in the callback) and all that would be read together anyway so it doesn't change the callback's running time to have them together.

Reply:
Ignore me.